packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,313 @@
+     */
+    @attr
+    public orientation: SliderOrientation = SliderOrientation.horizontal;
+    private orientationChanged(): void {
+        if (this.orientation === SliderOrientation.horizontal) {
+            this.classList.remove("slider-vertical");
+            this.classList.add("slider-horizontal");
+        } else {
+            this.classList.remove("slider-horizontal");
+            this.classList.add("slider-vertical");
+        }
+    }
+
+    /**
+     * mode value, default singleValue
+     */
+    @attr
+    public mode: SliderMode = SliderMode.singleValue;
+
+    protected proxy = document.createElement("input");
+
+    constructor() {
+        super();
+        this.proxy.setAttribute("type", "range");
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.direction = this.getDirection();
+        if (this.direction === Direction.rtl) {
+            this.value = `${this.value}`;
+        }
+        this.updateForm();
+        this.setupTrackConstraints();
+        this.setupListeners();
+        this.setupDefaultValue();
+    }
+
+    public disconnectedCallback(): void {
+        this.removeEventListener("keydown", this.keypressHandler);
+        this.removeEventListener("mousedown", this.clickHandler);
+        this.thumb.removeEventListener("mousedown", this.handleThumbMouseDown);
+    }
+
+    public increment = (): void => {
+        const newVal: number =
+            this.direction !== Direction.rtl &&
+            this.orientation !== SliderOrientation.vertical
+                ? Number(this.value) + Number(this.step)
+                : Number(this.value) - Number(this.step);
+        const incrementedVal: number = this.convertToConstrainedValue(newVal);
+        const incrementedValString: string =
+            incrementedVal < Number(this.max) ? `${incrementedVal}` : `${this.max}`;
+        this.value = incrementedValString;
+        this.updateForm();
+    };
+
+    public decrement = (): void => {
+        const newVal =
+            this.direction !== Direction.rtl &&
+            this.orientation !== SliderOrientation.vertical
+                ? Number(this.value) - Number(this.step)
+                : Number(this.value) + Number(this.step);
+        const decrementedVal: number = this.convertToConstrainedValue(newVal);
+        const decrementedValString: string =
+            decrementedVal > Number(this.min) ? `${decrementedVal}` : `${this.min}`;
+        this.value = decrementedValString;
+        this.updateForm();
+    };
+
+    protected keypressHandler = (e: KeyboardEvent) => {
+        super.keypressHandler(e);
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                this.increment();
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowDown:
+                this.decrement();
+                break;
+        }
+    };
```

Comment:
Should `shift + arrowRight` and others increment or decrement based on the step property?

Reply:
apparently not, fixed and added  `home` and `end` key processing